website/prebuild.py

```diff
@@ -142,15 +142,36 @@ def label_root_file(
     add_frontmatter(f, f.read_text(), [title], top_nav_order, has_children)
 
 
+def unlink_dir(path: Path) -> None:
+    # TODO: Use path.walk once we require Python 3.12.
+    for root, dirs, files in os.walk(path, topdown=False):
+        root_path = Path(root)
+        for name in files:
+            (root_path / name).unlink()
+        for name in dirs:
+            (root_path / name).rmdir()
+
+
 def main() -> None:
     # Ensure this runs from the repo root.
     os.chdir(Path(__file__).parents[1])
 
+    # bazel-generated symlinks interfere with jekyll because they may contain
+    # broken symlinks.
+    Path("bazel-out").unlink(missing_ok=True)
+    Path("bazel-bin").unlink(missing_ok=True)
+    Path("bazel-genfiles").unlink(missing_ok=True)
     # bazel-execroot interferes with jekyll because it's a broken symlink.
     Path("bazel-execroot").unlink()
     # This is a symlink to website/favicon.png, which is moved below.
     # TODO: Consider moving the icon to a location which won't break.
     Path("utils/vscode/images/icon.png").unlink()
+    # This may contain a README.md file that jekyll can't parse and that
+    # shouldn't appear in the navigation tree.
+    unlink_dir(Path("utils/vscode/node_modules"))
```

Comment:
I think we may accrue other node_modules. I think broken symlinks were a particular issue due to how jekyll works, it'd error too early. For this, had you considered in `_config.yml`:

```
exclude:
- **/node_modules
```

https://jekyllrb.com/docs/configuration/options/
https://ruby-doc.org/3.3.5/File.html#method-c-fnmatch

Note, I'm not sure whether this would also work for the .jekyll-cache dir, might make sense for both?

Reply:
That doesn't fix the problem -- the navigation tree building also breaks if the `node_modules` contains a `README.md` (as this one did for me). I think we could do that for some of the other things that only break in the jekyll step, not in the prebuild step, though.